packages/hoppscotch-app/components/http/Body.vue

```diff
@@ -80,19 +99,63 @@
   </div>
 </template>
 
-<script lang="ts">
-import { defineComponent } from "@nuxtjs/composition-api"
+<script setup lang="ts">
+import { watch, ref } from "@nuxtjs/composition-api"
+import { HoppRESTHeader } from "@hoppscotch/data"
+import { pipe } from "fp-ts/function"
+import * as A from "fp-ts/Array"
+import * as O from "fp-ts/Option"
+import { RequestOptionTabs } from "./RequestOptions.vue"
 import { useStream } from "~/helpers/utils/composables"
-import { restContentType$, setRESTContentType } from "~/newstore/RESTSession"
 import { knownContentTypes } from "~/helpers/utils/contenttypes"
+import {
+  restContentType$,
+  restHeaders$,
+  setRESTContentType,
+  setRESTHeaders,
+  addRESTHeader,
+} from "~/newstore/RESTSession"
 
-export default defineComponent({
-  setup() {
-    return {
-      validContentTypes: Object.keys(knownContentTypes),
+const emit = defineEmits<{
+  (e: "change-tab", value: string): void
+}>()
 
-      contentType: useStream(restContentType$, null, setRESTContentType),
-    }
+const validContentTypes = Object.keys(knownContentTypes)
+const contentType = useStream(restContentType$, null, setRESTContentType)
+
+// The functional headers list (the headers actually in the system)
+const headers = useStream(restHeaders$, [], setRESTHeaders)
+
+const overridenContentType = ref<string>("")
+
+watch(
+  headers,
+  (allHeaders: HoppRESTHeader[]) => {
+    overridenContentType.value = pipe(
+      allHeaders,
+      A.findFirst((h) => h.key.toLowerCase() === "content-type"),
```

Comment:
`overridenContentType` should check the `active` boolean value in headers entry.

Reply:
Please use `findLast()`
```suggestion
      A.findLast((h) => h.key.toLowerCase() === "content-type"),
```
Last occurance has highest preference if there're multiple entries with same key.